index.js

```diff
@@ -67,15 +67,19 @@ const matcher = (inputs, patterns, options, firstMatchOnly) => {
 
 	patterns = patterns.map(pattern => makeRegexp(pattern, options));
 
+	const {allPatterns} = options || {};
+	const didFit = new Uint8Array(patterns.length);
 	const result = [];
 
 	for (const input of inputs) {
 		let matches;
-		//	String is included only if it matchers at least one non-negated pattern supplied.
+		//	String is included only if it matches at least one non-negated pattern supplied.
+		//  NB: the `allPatterns` option requires every non-negated pattern to be matched once.
 		//	Matching a negated pattern excludes the string.
 
-		for (const pattern of patterns) {
+		for (let i = 0, pattern; (pattern = patterns[i]) !== undefined; i += 1) {
```

Comment:
I think you can still use a `for-of` loop. Just use `patterns.entries()`.

Reply:
Did some benchmarking once and found out the Node.js for(...of...) loop being much slower than indexed for(;;). Will do some fresh benchmarking today. I dont know how time critical most of the matcher use cases are.

Reply:
The difference looks quite substantial - take a look: https://github.com/valango/iterations-benchmarks
- so I'm looking forward for your decision.

Reply:
Micro-benchmarks are not very useful. They don't test actual real-world use-cases. I prefer we go with `for-of`.

Reply:
Agreed - they only give a hint. Having no better real-world use-case by hand, I used your benchmark fixture, see: https://gist.github.com/valango/e66ac27c6eec784f2d72d43438125131 .

On my machine, before the commit f49872c it gave avg 45 msecs; after the commit it gave 60. But the decision will be yours.